pandas/core/indexes/period.py

```diff
@@ -148,144 +177,176 @@ class PeriodIndex(PeriodArrayMixin, DatelikeOps, DatetimeIndexOpsMixin,
+        freq = kwargs.pop("freq", None)
+        if freq:
+            values = values.asfreq(freq)
+
+        attributes = self._get_attributes_dict()
+        attributes.update(kwargs)
+        if not len(values) and 'dtype' not in kwargs:
+            attributes['dtype'] = self.dtype
+        return self._simple_new(values, **attributes)
+
+    def _shallow_copy_with_infer(self, values=None, **kwargs):
         """ we always want to return a PeriodIndex """
         return self._shallow_copy(values=values, **kwargs)
 
-    def _coerce_scalar_to_index(self, item):
-        """
-        we need to coerce a scalar to a compat for our index type
+    # ------------------------------------------------------------------------
+    # Boxing
+    # err. maybe not...
 
-        Parameters
-        ----------
-        item : scalar item to coerce
+    @property
+    def _box_func(self):
+        def func(x):
+            if isinstance(x, Period) or x is tslib.NaT:
+                return x
+            else:
+                return Period._from_ordinal(ordinal=x, freq=self.freq)
+        return func
+
+    def _maybe_box_as_values(self, values, **attribs):
+        """Box an array of ordinals to a PeriodArray
+
+        This is purely for compatibility between PeriodIndex
+        and Datetime/TimedeltaIndex. Once these are all backed by
+        an ExtensionArray, this can be removed
         """
-        return PeriodIndex([item], **self._get_attributes_dict())
+        # TODO(DatetimeArray): remove
+        freq = attribs['freq']
+        return PeriodArray._from_ordinals(values, freq=freq)
+
+    # ------------------------------------------------------------------------
+    # Dispatch and maybe box. Not done in delegate_names because we box
+    # different from those (which use Index).
+
+    def asfreq(self, freq=None, how='E'):
+        result = self._data.asfreq(freq=freq, how=how)
+        return self._simple_new(result, name=self.name)
+
+    def _nat_new(self, box=True):
+        # TODO(DatetimeArray): remove this
+        result = self._data._nat_new(box=box)
+        if box:
+            result = self._simple_new(result, name=self.name)
+        return result
+
+    def to_timestamp(self, freq=None, how='start'):
+        from pandas import DatetimeIndex
+        result = self._data.to_timestamp(freq=freq, how=how)
+        return DatetimeIndex._simple_new(result,
+                                         name=self.name,
+                                         freq=result.freq)
+
+    def _format_native_types(self, na_rep=u'NaT', quoting=None, **kwargs):
+        # just dispatch, return ndarray
+        return self._data._format_native_types(na_rep=na_rep,
+                                               quoting=quoting,
+                                               **kwargs)
+
+    def _maybe_convert_timedelta(self, other):
+        # just dispatch, return ndarray
+        return self._data._maybe_convert_timedelta(other)
```

Comment:
Is there an overarching logic behind when you're using .values vs when ._data?  The latter I guess is slightly more performant, but the former seems easier to remember as "this is always lossless (or at least will be)"